warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py

```diff
@@ -227,10 +229,30 @@ def send_commands_from_testdata(self, testdatafile, **args):
                     result, response = new_obj_session._send_command_retrials(
                         details_dict, index=i, result=result,
                         response=response, system_name=td_sys)
-                    rspRes, response_dict = new_obj_session._get_response_dict(
-                        details_dict, i, response, response_dict)
-                    result = (result and rspRes) if "ERROR" not in (
-                                result, rspRes) else "ERROR"
+                    rspRes, resp_key_list = new_obj_session._get_response_dict(
+                        details_dict, i, response, resp_key_list)
+
+                    if resp_key_list:
+                        for resp in resp_key_list[i].keys():
+                            # session id is formed from the system name and session name.
+                            session_id = \
+                             self.get_session_id_for_resp_ref(details_dict, resp_key_list[i],
```

Comment:
@sugan19 
instead of writing a new method to get the session id that was used to send the command please update the existing method _get_obj_session to get the session_id in addition to the session_object.
Because the _get_obj_session is used to form the session_id to which we send the command but today it return the session_object and not the session_id.. but it does have the session_id information.
So please update _get_obj_session and wile updating do not modify the existing return statement but add an additional flag to request sesssion_id to be returned and if the flag is set add a new return statement to return session_id as well.

We should not be forming session_id's in two places , it will be difficult to track changes.. so it is best to update the method that forms the session_id to which the command is sent to...

Reply:
@satyakrish Addressed you review comments